digestparser/json_output.py

```diff
@@ -0,0 +1,183 @@
+"build JSON output from digest content"
+import os
+import time
+import copy
+from collections import OrderedDict
+import requests
+from elifetools.utils import copy_attribute
+from digestparser.utils import msid_from_doi
+from digestparser.jats import (parse_jats_file, parse_jats_digest, parse_jats_pub_date,
+                               parse_jats_subjects, xml_to_html)
+from digestparser.build import build_digest
+
+
+def iiif_server_info(info_url):
+    "get the image info from the IIIF server"
+    info = {}
+    try:
+        response = requests.get(info_url)
+        info = response.json()
+    except:
+        # could be any error right now
+        pass
+    return info
+
+
+def image_info(msid, file_name, digest_config):
+    "get image info from the IIIF server"
+    if not msid or not file_name or not digest_config:
```

Comment:
I might do something else here, because it would be helpful to be able to produce some JSON even if there is no image, for viewing, testing, or adding image data as a separate step. I think it is too strict to assume it always has an image, which is how it is now.

Reply:
That need should lead us to cleanly separate the code for the image data (as you say `as a separate step` A) from the other fields (step B) of the digest JSON; so tests, scripts, etc. should be able to execute just step B, or just step A, and you don't have to stub the image and its HTTP calls all the time.

We could call these parts `digest image` and ~`digest data`~ ~`digest text`~ ~`digest content`~ `digest text version`|`digest text only`. Alternatively, we could generalize and say a `digest` is made up of multiple `digest parts` but still have to name the parts.

Reply:
I committed my vision of it with two additional test scenarios if you do not supply all the parts (JATS file and image file), which produces less complete JSON output.

Reply:
https://github.com/elifesciences/digest-parser/pull/43/commits/f5259f4ae359cdb796f31811935762384a3083b9